sdk/modelsrepository/Azure.Iot.ModelsRepository/src/ModelQuery.cs

```diff
@@ -0,0 +1,176 @@
+            };
+        }
+
+        public ModelMetadata GetMetadata()
+        {
+            return new ModelMetadata(GetId(), GetExtends(), GetComponentSchemas());
+        }
+
+        public string GetId()
+        {
+            using (JsonDocument document = JsonDocument.Parse(_content, _parseOptions))
+            {
+                JsonElement _root = document.RootElement;
+
+                if (_root.ValueKind == JsonValueKind.Object && _root.TryGetProperty("@id", out JsonElement id))
+                {
+                    if (id.ValueKind == JsonValueKind.String)
+                    {
+                        return id.GetString();
+                    }
+                }
+            }
+
+            return string.Empty;
+        }
+
+        public IList<string> GetExtends()
+        {
+            List<string> dependencies = new List<string>();
+
+            using (JsonDocument document = JsonDocument.Parse(_content, _parseOptions))
+            {
+                JsonElement _root = document.RootElement;
+
+                if (_root.ValueKind == JsonValueKind.Object && _root.TryGetProperty("extends", out JsonElement extends))
+                {
+                    if (extends.ValueKind == JsonValueKind.Array)
+                    {
+                        foreach (JsonElement extendElement in extends.EnumerateArray())
+                        {
+                            if (extendElement.ValueKind == JsonValueKind.String)
+                            {
+                                dependencies.Add(extendElement.GetString());
+                            }
+                            else if (extendElement.ValueKind == JsonValueKind.Object)
+                            {
+                                // extends can have multiple levels and can contain components.
+                                // TODO: Support object ctor - inefficient serialize.
+                                ModelMetadata nested_interface = new ModelQuery(JsonSerializer.Serialize(extendElement)).GetMetadata();
+                                dependencies.AddRange(nested_interface.Dependencies);
+                            }
+                        }
+                    }
+                    else if (extends.ValueKind == JsonValueKind.String)
+                    {
+                        dependencies.Add(extends.GetString());
+                    }
+                }
+            }
+
+            return dependencies;
+        }
+
+        // TODO: Consider refactor to an object type based processing.
+        public IList<string> GetComponentSchemas()
+        {
+            List<string> componentSchemas = new List<string>();
+
+            using (JsonDocument document = JsonDocument.Parse(_content, _parseOptions))
+            {
+                JsonElement _root = document.RootElement;
+
+                if (_root.ValueKind == JsonValueKind.Object && _root.TryGetProperty("contents", out JsonElement contents))
```

Comment:
The nesting below is nuts and very hard to read. Probably needs rework.

Reply:
task added.